Library/DiscUtils.Core/Internal/Utilities.cs

```diff
@@ -676,53 +676,26 @@ public static string CombinePaths(string a, string b)
         /// </summary>
         /// <param name="basePath">The base path to resolve from.</param>
         /// <param name="relativePath">The relative path.</param>
-        /// <returns>The absolute path, so far as it can be resolved.  If the
-        /// <paramref name="relativePath"/> contains more '..' characters than the
-        /// base path contains levels of directory, the resultant string will be relative.
-        /// For example: (TEMP\Foo.txt, ..\..\Bar.txt) gives (..\Bar.txt).</returns>
+        /// <returns>The absolute path. If no <paramref name="basePath"/> is specified
+        /// then relativePath is returned as-is. If <paramref name="relativePath"/>
+        /// contains more '..' characters than the base path contains levels of 
+        /// directory, the resultant string be the root drive followed by the file name.
+        /// If no the basePath starts with '\' (no drive specified) then the returned
+        /// path will also start with '\'.
+        /// For example: (\TEMP\Foo.txt, ..\..\Bar.txt) gives (\Bar.txt).
+        /// </returns>
         public static string ResolveRelativePath(string basePath, string relativePath)
         {
-            List<string> pathElements =
-                new List<string>(basePath.Split(new[] { '\\' }, StringSplitOptions.RemoveEmptyEntries));
-            if (!basePath.EndsWith(@"\", StringComparison.Ordinal) && pathElements.Count > 0)
-            {
-                pathElements.RemoveAt(pathElements.Count - 1);
-            }
-
-            pathElements.AddRange(relativePath.Split(new[] { '\\' }, StringSplitOptions.RemoveEmptyEntries));
-
-            int pos = 1;
-            while (pos < pathElements.Count)
+            if (string.IsNullOrEmpty(basePath))
             {
-                if (pathElements[pos] == ".")
-                {
-                    pathElements.RemoveAt(pos);
-                }
-                else if (pathElements[pos] == ".." && pos > 0 && pathElements[pos - 1][0] != '.')
-                {
-                    pathElements.RemoveAt(pos);
-                    pathElements.RemoveAt(pos - 1);
-                    pos--;
-                }
-                else
-                {
-                    pos++;
-                }
+                return relativePath;
             }
 
-            string merged = string.Join(@"\", pathElements.ToArray());
-            if (relativePath.EndsWith(@"\", StringComparison.Ordinal))
-            {
-                merged += @"\";
-            }
+            string merged = Path.GetFullPath(Path.Combine(basePath, relativePath));
 
-            if (basePath.StartsWith(@"\\", StringComparison.Ordinal))
-            {
-                merged = @"\\" + merged;
-            }
-            else if (basePath.StartsWith(@"\", StringComparison.Ordinal))
+            if (basePath.StartsWith(@"\") && merged[1].Equals(':'))
```

Comment:
It's a lot shorter because it uses `Path.GetFullPath` to do most of the work instead of doing the work in custom code - so that's what you're seeing here. The only difference with `Path.GetFullPath` is that the root is always `\` instead of e.g. `C:\`.